test/intfuncs.jl

```diff
@@ -11,6 +11,12 @@
 
 @test typeof(gcd(Int32(3), Int32(15))) == Int32
 
+@test gcd(typemax(Int), 1) == 1
+@test gcd(-typemax(Int), 1) == 1
+@test gcd(typemin(Int), 1) == 1
+divex = sizeof(Int) == 4 ? OverflowError : InexactError
+@test_throws divex gcd(typemin(Int), typemin(Int))
```

Comment:
or if we don't use `checked_abs` in `gcd{T<:Union{Int64,UInt64,Int128,UInt128}}`, how about
```
@test_throws OverflowError gcd(typemin(Int32), typemin(Int32))
@test_throws InexactError gcd(typemin(Int64), typemin(Int64))
```
and run both everywhere?